### PR TITLE
Initial Security Considerations

### PR DESCRIPTION
**Pull request by jsalowey**
closes issue #5

### CODE REVIEW

File: draft-salowey-wimse-arch.md [hunk 1]
```diff
@@ -61,8 +61,23 @@ TODO Introduction
 
 # Security Considerations
 
-# Threat Model
-TODO Security
+## Threat Model
+
+### Traffic Interception
+
+Workloads communicating within an an applications may face different threats to traffic interception in different deployments. In many deployments security controls are deployed for internal communications at lower layers to reduce the risk of traffic observation and modification for network communications. When a security layer such as TLS is deployed in these environments the protection is usually hop-by-hop rather than end-to-end. Where TLS or other protection mechanisms terminate, information is left in the clear and information may be disclosed or modified.
```
Comment (yaroslavros) [inline draft-salowey-wimse-arch.md:68]: "hop-by-hop" and "end-to-end" mean wildly different things to different people.

I would suggest "When a security layer such as TLS is deployed in this environments in combination with TLS decryption middleware such as load balancers, gateways, proxies, and firewalls, the protection is provided only between each adjacent pair of TLS endpoints. There are no guarantees of correct identity passthrough in those middleware TLS interception devices and services."

File: draft-salowey-wimse-arch.md [hunk 2]
```diff
@@ -61,8 +61,23 @@ TODO Introduction
...
+
+### Information Disclosure
+
+Observation and interception of network traffic is not the only means of disclosure in these systems. Other vectors of information leakage is through disclosure in log files and other observability and troubleshooting mechanisms. For example, an application may log the contents of HTTP headers containing JWT bearer tokens.  The information in this logs may be made available to other systems with less stringent access controls which may result in this token falling into an attackers hands who then uses it to compromise a system.
```
Comment (yaroslavros) [inline draft-salowey-wimse-arch.md:72]: I would add:
This creates privacy risks and potential surface for reconnaissance attacks. If observed tokens can be reused, this also may allow attackers to impersonate workloads.

**Review by yaroslavros — COMMENTED** [review]
Few minor comments.

Again, great start!

File: draft-salowey-wimse-arch.md (hunk 1 above)
Comment (jsalowey) [reply draft-salowey-wimse-arch.md]: I made an attempt and incorporating this change.

File: draft-salowey-wimse-arch.md (hunk 2 above)
Comment (jsalowey) [reply draft-salowey-wimse-arch.md]: Added

**Pull request merged**